File: pe/submissions/accepted/bit.cpp
#include <bits/stdc++.h>
constexpr int N(1e6);
using namespace std;

int n, bit[N + 1];
void update(int i) {
    for (; i <= n; i += i & -i)
        ++bit[i];
}

int query(int i) {
    int ans = 0;
    for (; i; i -= i & -i)
        ans += bit[i];

    return ans;
}

int v[N + 1];
int main() {
    cin >> n;

    int i, x;
    for (i = 1; i <= n; ++i) {
        cin >> x;
        v[x] = i;
    }

    auto ans = (long long)n * (n - 1) / 2;
    for (i = 1; i <= n; ++i) {
        cin >> x;
        ans -= query(v[x]);
        update(v[x]);
    }

    cout << ans << '\n';
    return 0;
}

File: pe/submissions/accepted/bit.py
import sys

def main():
    input = sys.stdin.readline
    n = int(input())
    v = [0] * (n + 2)
    bit = [0] * (n + 2)

    # Read first permutation and store positions
    tokens = []
    while len(tokens) < n:
        tokens += list(map(int, input().split()))
    for i, x in enumerate(tokens, 1):
        v[x] = i

    # Fenwick Tree functions (1-based)
    def update(i):
        while i <= n:
            bit[i] += 1
            i += i & -i

    def query(i):
        res = 0
        while i:
            res += bit[i]
            i -= i & -i
        return res

    # Read second permutation and compute answer
    ans = n * (n - 1) // 2
    tokens = []
    while len(tokens) < n:
        tokens += list(map(int, input().split()))
    for x in tokens:
        ans -= query(v[x])
        update(v[x])

    print(ans)

if __name__ == '__main__':
    main()


#CPU time: 1.078 seconds
#Peak memory usage: 145.97 MiB